<a href="https://colab.research.google.com/github/afreire1174-png/afreire1174/blob/main/MVP_Alexandre_Freire_BD_Acidentes_v6.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# MVP — Machine Learning & Analytics

**Nome:** Alexandre Freire Magalhães  
**Matrícula:**xxxxxx
**Data:** Junho/2026  
**Dataset:** Base de Dados de acidentes do segmento de Geração e Transmissão de Energia  
**Tipo de problema:** Classificação binária  

---

## Título do projeto

# Predição de Acidentes Graves e Fatais em Empresas do Setor Elétrico utilizando Machine Learning





##1. DEFINIÇÃO DO PROBLEMA
1.1 Descrição do problema

O setor elétrico brasileiro apresenta atividades com elevado potencial de risco, especialmente nos segmentos de geração e transmissão de energia elétrica. Apesar da existência de procedimentos de segurança, treinamentos e controles operacionais, ainda ocorrem acidentes de trabalho com diferentes níveis de gravidade, incluindo eventos críticos com consequências severas para trabalhadores, ativos e continuidade operacional.

Este trabalho também terá por base os conceitos do modelo denominado de "Pirâmide de Bird", o qual estabelece que acidentes graves e fatais são precedidos por uma grande quantidade de incidentes de menor severidade e quase-acidentes, mas que apresentam características semelhantes. Este modelo estabelece que, para cada 1 acidente grave ou fatal, ocorrem aproximadamente 10 acidentes leves, 30 acidentes com danos materiais e 600 incidentes ou quase-acidentes. Com base nesse princípio, técnicas de Machine Learning podem analisar dados históricos de incidentes, quase-acidentes e acidentes para identificar padrões de risco e prever situações com maior probabilidade de evoluir para eventos de alto ou crítico grau de risco, apoiando ações preventivas e fortalecendo a segurança operacional no setor elétrico.

O impacto quando da ocorrência de acidentes graves e fatais afeta toda a cadeia produtiva das empresas do setor elétrico, envolvendo desde a alta direção, engenheiros e técnicos de saúde e segurança do trabalho (SST), equipes de planejamento das atividades, equipes de operação e manutenção (O&M) dos segmentos de geração e transmissão de energia elétrica, além das empresas contratadas para a execução de determinadas atividades do setor. Estes atores precisam ter ferramentas eficazes que possibilitem a eliminação ou mitigação dos riscos inerentes as atividades do setor elétrico, permitindo identificar situações com maior probabilidade de evoluir para acidentes graves ou fatais e direcionar ações preventivas antes da execução dos serviços.

Devido aos temas estarem altamente correlacionados a prevenção de acidentes, enumeramos a seguir a relevância por cada tema abordado neste MVP, a saber:

	• Engenharia de Segurança: aplicação prática na prevenção de acidentes ocupacionais.
	• Pirâmide de Bird: utilização dos quase-acidentes como indicadores preditivos de eventos mais graves.
	• Machine Learning: identificação automática de padrões de risco em grandes volumes de dados históricos.
	• Prevenção de fatalidades: foco na redução de acidentes críticos por meio de decisões preventivas orientadas por dados.

Por fim, devido a tudo que foi comentado neste item, verifica-se que este tema é de grande relevância para as empresas, não só do setor elétrico, como de outros setores também, pois além de evitar o risco reputacional, expondo negativamente a imagem das empresas no mercado, acarretando muitas vezes em desvalorização das ações na bolsa de valores, além das perdas financeiras por conta de indenizações as famílias dos trabalhadores envolvidos. Mas vale mencionar que a justificativa mais nobre deste trabalho é evitar perdas de vidas humanas em decorrência de acidentes de trabalhos que muitas vezes poderiam ser evitados com uma simples adoção de medidas de prevenção.


## 1.2 Objetivo do MVP

Construir e avaliar modelos de Machine Learning capazes de prever a ocorrência de acidentes classificados como de Alto ou Crítico Grau de Risco, utilizando informações históricas dos acidentes e quase-acidentes registrados na base de dados, contribuindo para a prevenção de acidentes graves e fatais.

Os modelos desenvolvidos possuem caráter de apoio à decisão, não substituindo a análise técnica realizada pelos profissionais de Engenharia de Segurança do Trabalho. Os resultados obtidos são válidos apenas para o conjunto de dados utilizado e podem não representar integralmente outros contextos ou organizações do setor elétrico.


## **1.2.1 Pergunta de Negócio**

É possível prever, a partir das características históricas dos acidentes e quase-acidentes do setor elétrico, quais ocorrências possuem maior probabilidade de serem classificadas como de Alto ou Crítico Grau de Risco?


## 1.2.2 Objetivos Específicos

	• Realizar análise exploratória dos dados.
	• Preparar a base para Machine Learning.
	• Tratar valores ausentes e inconsistências.
	• Codificar variáveis categóricas.
	• Tratar eventual desbalanceamento das classes.
	• Treinar modelos de classificação.
	• Identificar as variáveis mais relevantes para a previsão.

**1.3 Tipo de problema**

	Tipo escolhido: Classificação Binária

	Variável alvo:

	Evento_Crítico_Target
		SIM:
			Grau de risco Alto
			Grau de risco Crítico
		NÃO:
			Grau de risco Baixo
			Grau de risco Médio

O problema proposto neste MVP caracteriza-se como um problema de classificação supervisionada, pois a variável alvo da base de dados, Evento_Crítico_Target, é categórica e assume apenas duas classes possíveis: "Sim" e "Não".

A classe "Sim" representa ocorrências classificadas como eventos de Alto ou Crítico Grau de Risco, enquanto a classe "Não" representa ocorrências que não se enquadram nessa classificação. Assim, o objetivo do modelo é aprender, a partir das características históricas dos acidentes e quase-acidentes do setor elétrico, a identificar corretamente a qual dessas duas categorias uma nova ocorrência pertence.

Como a variável resposta não representa um valor numérico contínuo, mas sim categorias mutuamente exclusivas, o problema não é de regressão. Em vez de estimar uma quantidade, o modelo deverá decidir entre duas classes previamente definidas, caracterizando uma classificação binária supervisionada.

Além disso, a base de dados já possui registros históricos rotulados com a variável Evento_Crítico_Target, permitindo que os algoritmos aprendam padrões presentes nas variáveis preditoras — como segmento, região, agente causador, potencial do acidente e tempo de empresa — para posteriormente classificar novas ocorrências como "Sim" ou "Não".

O objetivo dos modelos será prever se determinado registro possui potencial para ser classificado como evento crítico.




**1.4 Premissas, hipóteses e critérios de sucesso**

**Premissas adotadas:**

	• A base de dados histórica representa adequadamente as ocorrências de acidentes e quase-acidentes do setor elétrico.
	• As variáveis disponíveis são relevantes para prever eventos de Alto ou Crítico Grau de Risco.
	• O processo de preparação dos dados (tratamento de valores ausentes, codificação das variáveis categóricas e balanceamento das classes, quando necessário) será suficiente para possibilitar o treinamento dos modelos.
	• A Pirâmide de Bird é aplicável ao contexto do setor elétrico e fundamenta a identificação preventiva de eventos críticos.

**Hipóteses adotas:**

	• É possível utilizar Machine Learning para prever ocorrências de Alto ou Crítico Grau de Risco a partir dos dados históricos de acidentes e quase-acidentes.
	• Variáveis como potencial do acidente, agente causador, segmento e região influenciam a classificação do evento.
	• Modelos baseados em árvores de decisão (como Random Forest e XGBoost) tendem a apresentar melhor desempenho na classificação dos eventos críticos.
	• A identificação antecipada de eventos críticos pode apoiar ações preventivas e contribuir para a redução de acidentes graves e fatais.

**Critérios adotados:**

	• Desenvolver um pipeline completo de Machine Learning, desde a preparação dos dados até a avaliação dos modelos.
	• Obtenção de um modelo com desempenho satisfatório na identificação de ocorrências de Alto ou Crítico Grau de Risco, priorizando a métrica Recall, por ser mais importante reduzir a ocorrência de falsos negativos (eventos críticos classificados incorretamente como não críticos).
	• Comparar o modelo baseline com modelos candidatos utilizando as métricas Recall, F1-Score e ROC-AUC.
	• Identificar as variáveis mais relevantes para a previsão dos eventos críticos.
	• Apresentar uma análise crítica dos resultados, incluindo limitações e oportunidades de melhoria.

**2. Ambiente, bibliotecas e reprodutibilidade**

O desenvolvimento deste MVP foi realizado em Python 3.x, utilizando bibliotecas amplamente empregadas em projetos de Ciência de Dados e Machine Learning.

### As principais bibliotecas utilizadas foram:
	• Pandas: manipulação e preparação dos dados;
	• NumPy: operações numéricas;
	• Scikit-learn (sklearn): pré-processamento dos dados, divisão entre treinamento e teste, construção dos modelos de Machine Learning e avaliação de desempenho;
	• Matplotlib e Seaborn (quando utilizadas): visualização gráfica dos dados e dos resultados.

Para garantir a reprodutibilidade dos experimentos, foi definida uma semente fixa (random_state = 42) em todas as etapas que envolvem aleatoriedade, como a divisão dos dados em treinamento e teste e o treinamento dos modelos.


Como os modelos utilizados (Regressão Logística e Random Forest) possuem baixo custo computacional para a base de dados analisada, o processamento foi realizado em CPU, não sendo necessária a utilização de GPU.



**Código desta seção encontra-se a seguir:**


In [6]:
# ======================
# 2. Ambiente e bibliotecas
# ======================

import os
import sys
import random
import warnings
import seaborn as sns

import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

from sklearn.model_selection import train_test_split, StratifiedKFold, RandomizedSearchCV
from sklearn.compose import ColumnTransformer
from sklearn.pipeline import Pipeline
from sklearn.preprocessing import OneHotEncoder, StandardScaler
from sklearn.impute import SimpleImputer

from sklearn.dummy import DummyClassifier
from sklearn.linear_model import LogisticRegression
from sklearn.ensemble import RandomForestClassifier

from sklearn.metrics import (
 accuracy_score,
 precision_score,
 recall_score,
 f1_score,
 roc_auc_score,
 classification_report,
 confusion_matrix,
 ConfusionMatrixDisplay
)

from scipy.stats import randint

warnings.filterwarnings("ignore")

SEED = 42
np.random.seed(SEED)
random.seed(SEED)

print("Python:", sys.version.split()[0])
print("Seed:", SEED)
print("NumPy:", np.__version__)
print("Pandas:", pd.__version__)

import sklearn
print("Scikit-learn:", sklearn.__version__)

Python: 3.12.13
Seed: 42
NumPy: 2.0.2
Pandas: 2.2.2
Scikit-learn: 1.6.1


### **2.1 Dependências adicionais**

Além das bibliotecas de Ciência de Dados e Machine Learning, foram utilizadas algumas dependências auxiliares para garantir a organização, execução e reprodutibilidade do projeto:


		○ os: manipulação de caminhos e arquivos do sistema operacional;
		○ sys: obtenção de informações sobre a versão do Python utilizada no ambiente de execução;
		○ random: geração de números aleatórios com semente fixa para garantir a reprodutibilidade dos experimentos;
		○ warnings: supressão de mensagens de aviso que não interferem na execução do notebook, tornando a saída mais limpa.

Não foi necessária a instalação de bibliotecas adicionais além das disponíveis no ecossistema científico padrão do Python, como NumPy, Pandas, Matplotlib, Seaborn e Scikit-learn.


**O Código abaixo documenta as dependências adicionais e alinhada ao projeto de classificação em Machine Learning:**


In [7]:
		import os
		import sys
		import random
		import warnings

		warnings.filterwarnings("ignore")

		SEED = 42
		random.seed(SEED)

		print("Python:", sys.version)


Python: 3.12.13 (main, Mar  4 2026, 09:23:07) [GCC 11.4.0]


### **2.2 Funções auxiliares**

Esta seção contém apenas funções reutilizáveis que facilitem a análise e a avaliação dos modelos, deixando-os mais organizado, enxuto e reutilizável.

**Código a seguir:**

In [8]:
	# ==========================================
	# Funções auxiliares para avaliação dos modelos
	# ==========================================

	def avaliar_modelo(nome_modelo, y_true, y_pred, tempo_treino=None):
	 """
	 Calcula as principais métricas de classificação para o problema
	 de previsão de ocorrências de Alto ou Crítico Grau de Risco.
	 """

	 return {
	 "Modelo": nome_modelo,
	 "Accuracy": accuracy_score(y_true, y_pred),
	 "Precision": precision_score(y_true, y_pred, pos_label=1, zero_division=0),
	 "Recall": recall_score(y_true, y_pred, pos_label=1, zero_division=0),
	 "F1-Score": f1_score(y_true, y_pred, pos_label=1, zero_division=0),
	 "Tempo de treino (s)": tempo_treino
	 }


	def plotar_matriz_confusao(y_true, y_pred):
	 """
	 Exibe a matriz de confusão para o problema de classificação.
	 """

	 labels = [0, 1]
	 nomes_classes = ["Baixo/Médio", "Alto/Crítico"]

	 cm = confusion_matrix(y_true, y_pred, labels=labels)

	 disp = ConfusionMatrixDisplay(
	 confusion_matrix=cm,
	 display_labels=nomes_classes
	 )

	 disp.plot(values_format="d")
	 plt.title("Matriz de Confusão")
	 plt.show()


	def obter_colunas_existentes(df, colunas_candidatas):
	 """
	 Retorna apenas as colunas existentes no DataFrame.
	 """

	 return [col for col in colunas_candidatas if col in df.columns]


	def salvar_figura(nome_arquivo):
	 """
	 Salva a figura atual em alta resolução.
	 """

	 plt.savefig(nome_arquivo, dpi=300, bbox_inches="tight")


### **3. Seleção e carga dos dados**

**3.1 Fonte dos dados**

Informações gerais sobre a fonte de dados:

	• Nome do dataset: Histórico de Incidentes das atividades dos segmentos de Geração e Tranamissão_2023 a 2026;

	• Link da fonte: Antes da entrega, publique o arquivo base_ml_mvp1_acidentes.csv em um repositório público do GitHub e substitua a variável DATA_URL abaixo pelo link raw.githubusercontent.com.

	• Por que esse dataset foi escolhido: Esse dataset foi escolhido por apresentar o histórico dos acidentes e quase acidentes ocorridos nos segmentos de Geração e Transmissão entre 2023 e 2026, além de outras variáveis importantes para a previsão de acidentes classificados como de Alto ou Crítico Grau de Risco;

	• quais restrições ou condições foram consideradas: LGPD, limitação da base de dados, qualidade dos dados (registros incompletos/inconsistentes), desbalanceamento das classes (uma vez que ocorrências de Alto ou Crítico Grau de Risco tendem a ser menos frequentes do que eventos de menor gravidade) e limitação de generalização do modelo;

	• se há questões de ética, privacidade, confidencialidade ou licença: As informações mais sensíveis deste histórico foram expurgadas conforme estabelece a Lei Geral de Proteção de Dados (LGPD) . Podemos citar como exemplo de tais informações as seguintes: nome, endereço e documentação do colaborador envolvido nos acidentes, dados médicos, dados oriundos das investigações dos acidentes, bem como o nome das empresas envolvido tendo por base as políticas de compliance de cada uma dessas empresas.


### **3.2 Carga dos dados**

A célula abaixo carrega os dados a partir de uma URL pública. Para facilitar seus testes locais, há uma alternativa de leitura do arquivo CSV incluído no pacote, mas a versão final para entrega deve usar a URL pública.

